ament_cmake_auto/cmake/ament_auto_find_build_dependencies.cmake

```diff
@@ -27,18 +32,40 @@
 # @public
 #
 macro(ament_auto_find_build_dependencies)
-  if(ARGN)
+  cmake_parse_arguments(_ARG "" "" "REQUIRED" ${ARGN})
+  if(_ARG_UNPARSED_ARGUMENTS)
     message(FATAL_ERROR "ament_auto_find_build_dependencies() called with "
-      "unused arguments: ${ARGN}")
+      "unused arguments: ${_ARG_UNPARSED_ARGUMENTS}")
   endif()
 
   if(NOT _AMENT_PACKAGE_NAME)
     ament_package_xml()
   endif()
 
+  # ensure that the caller isn't expecting additional packages to be found
+  set(_build_and_buildtool_depends
+    ${${PROJECT_NAME}_BUILD_DEPENDS}
+    ${${PROJECT_NAME}_BUILDTOOL_DEPENDS}
+  )
+  set(_ignored_packages "")
```

Comment:
Nitpick: "ignored" doesn't seem to match what is happening with these. The package names are not being ignored but provided additionally resulting in a fatal error. Therefore I would suggest choosing a different variable name. Maybe `_additional_packages`?